### ISSUE TITLE
source links in the References sidebar go out of date easily

### ISSUE THREAD

**Issue by skybrian** [dart-atom/dartlang#302]
I keep the References sidebar open while updating all the references to a field. After I make a few edits at the top of the file, all the source links point to the wrong character ranges.

A quick fix would be to add a refresh button. Alternately, perhaps automatically re-run the search, though perhaps that's expensive. Or perhaps, the links need to represented in a different way that doesn't go out of date quite so easily when unrelated parts of the file are edited.

**Comment by devoncarew** [comment]
Hmm, interesting. A refresh button wouldn't be bad, esp. if it had some use case other than just the tool lets the refs go out of date. Like if the user was making changes based on the search results and wanted to see a shorter list of the remaining results.

Atom lets you create markers in a file, and will keep those markers up to date with changes. When we the current file corresponds to some results on the references view, we could create atom markers for those matches. We'd then listen for changes to them, and remove the markers when the active file changed.

It's a bit more machinery than we currently have in the references view though -

**Comment by skybrian** [comment]
Yep, a refresh button would work for me, and re-running the search to see what remains is something I do sometimes.
